project/src/main/java/com/google/autograder/servlets/ListCoursesServlet.java

```diff
@@ -44,7 +64,19 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
 
                 String json = API.getJSON(connection);
 
-                response.getWriter().println(json);
+                // System.out.println(json);
+
+                // System.out.println("\n\n\n");
+
+                storeCourseDataInDatastore(json);
+
+                String coursesData = getCoursesDataFromDatastore();
```

Comment:
An improvement would be:
1. ListCourses from Classroom api
2. GetCourses data from data store.
3. Compare the result, if different, then call storeCourseDataInDatastore.

You can add a follow up PR for this improvement.

Reply:
Yeah that makes sense, I'm going to get to that soon. I wanted to get the flow working from Course -> Assignment -> Submissions working first before I optimized it. I also wanted to wait until after I integrated with Mahima's larger Database class.